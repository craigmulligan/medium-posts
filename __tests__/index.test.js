const { getUser, getPosts } = require('../')

describe('.getUser', () => {
  test('should return user with properties', async () => {
    const user = await getUser()
    expect(user).toMatchSnapshot()
  })
})

describe('.getPosts', () => {
  test('should return posts with limit', async () => {
    const posts = await getPosts('the_economist', 30)
    expect(posts.length).toBe(30)
  })

  test('should return posts with url', async () => {
    const posts = await getPosts('the_economist')
    const expected = [
      expect.objectContaining({
        url: expect.any(String),
        image: expect.any(String)
      })
    ]
    expect(posts).toEqual(expect.arrayContaining(expected))
  })
})
