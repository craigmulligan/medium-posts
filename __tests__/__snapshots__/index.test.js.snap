// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.getUser should return user with properties 1`] = `
Object {
  "allowNotes": 1,
  "backgroundImageId": "",
  "bio": "",
  "createdAt": 1462711206107,
  "facebookAccountId": "10153358094910666",
  "imageId": "0*UpyzJf7LPK3u_tIh.",
  "isNsfw": false,
  "lastPostCreatedAt": 1515117924656,
  "mediumMemberAt": 0,
  "name": "Craig Andrew Mulligan",
  "twitterScreenName": "",
  "type": "User",
  "userId": "38725f58c664",
  "username": "hobochild",
}
`;
